core/src/main/java/org/apache/carbondata/locks/HdfsFileLock.java

```diff
@@ -108,7 +109,9 @@ public boolean unlock() {
       } catch (IOException e) {
         return false;
       } finally {
-        if (FileFactory.getCarbonFile(location, FileFactory.getFileType(location)).delete()) {
+        CarbonFile carbonFile =
+            FileFactory.getCarbonFile(location, FileFactory.getFileType(location));
+        if (carbonFile.exists() && carbonFile.delete()) {
           LOGGER.info("Deleted the lock file " + location);
         } else {
           LOGGER.error("Not able to delete the lock file " + location);
```

Comment:
Please handle it differently if the file is not exist. This log gives wrong indication.